controllers/yawol-controller/loadbalancer/loadbalancer_controller_test.go

```diff
@@ -248,6 +250,93 @@ var _ = Describe("loadbalancer controller", Serial, Ordered, func() {
 			})
 		})
 
+		It("should scale down old lbset after new one has ready keepalived", func() {
+			var oldLbs runtimeClient.ObjectKey
+			By("waiting for lbset creation")
+			hopefully(lbNN, func(g Gomega, act LB) error {
+				var lbsetList yawolv1beta1.LoadBalancerSetList
+				g.Expect(k8sClient.List(ctx, &lbsetList, &runtimeClient.ListOptions{
+					LabelSelector: labels.SelectorFromSet(lb.Spec.Selector.MatchLabels),
+				})).Should(Succeed())
+
+				g.Expect(len(lbsetList.Items)).Should(Equal(1))
+				oldLbs = runtimeClient.ObjectKeyFromObject(&lbsetList.Items[0])
+				return nil
+			})
+
+			By("changing flavorid")
+			updateLB(lbNN, func(act *LB) {
+				act.Spec.Infrastructure.Flavor = yawolv1beta1.OpenstackFlavorRef{
+					FlavorID: pointer.String("somenewid"),
+				}
+			})
+
+			var newLbs runtimeClient.ObjectKey
+			By("checking for a new lbset")
+			hopefully(lbNN, func(g Gomega, act LB) error {
+				var lbsetList yawolv1beta1.LoadBalancerSetList
+				g.Expect(k8sClient.List(ctx, &lbsetList, &runtimeClient.ListOptions{
+					LabelSelector: labels.SelectorFromSet(lb.Spec.Selector.MatchLabels),
+				})).Should(Succeed())
+
+				g.Expect(len(lbsetList.Items)).Should(Equal(2))
+				for i := range lbsetList.Items {
+					if lbsetList.Items[i].Annotations[helper.RevisionAnnotation] == "2" {
+						newLbs = runtimeClient.ObjectKeyFromObject(&lbsetList.Items[i])
+					}
+				}
+				return nil
+			})
+
+			By("Make both lbsets available by path status", func() {
+				var lbsetList yawolv1beta1.LoadBalancerSetList
+				Expect(k8sClient.List(ctx, &lbsetList, &runtimeClient.ListOptions{
+					LabelSelector: labels.SelectorFromSet(lb.Spec.Selector.MatchLabels),
+				})).Should(Succeed())
+				for _, lbs := range lbsetList.Items {
+					lbs.Status.ReadyReplicas = &lbs.Spec.Replicas
+					lbs.Status.Replicas = &lbs.Spec.Replicas
+					Expect(k8sClient.Status().Update(ctx, &lbs)).Should(Succeed())
```

Comment:
I suggest using a patch here (without optimistic locking). Otherwise, this is likely to cause flakes because of conflict errors.